packages/sanity/src/presentation/preview/Preview.tsx

```diff
@@ -265,19 +266,135 @@ export const Preview = memo(
           'startViewTransition' in document &&
           typeof document.startViewTransition === 'function'
         ) {
-          document.startViewTransition(() => flushSync(() => update()))
+          document.startViewTransition({
+            // @ts-expect-error - fix typings
+            update: () => flushSync(() => update()),
+            types: ['sanity-iframe-viewport'],
+          })
         } else {
           update()
         }
       }
     }, [canUseViewTransition, prefersReducedMotion, currentViewport, viewport])
 
+    const toast = useToast()
+    const allowOrigins = useAllowPatterns(previewUrlRef)
+    const [checkOrigin, setCheckOrigin] = useState<false | string>(false)
+    const [reportedMismatches] = useState(new Set<string>())
+    const reportMismatchingOrigin = useEffectEvent((reportedOrigin: string) => {
+      if (allowOrigins.some((allow) => allow.test(reportedOrigin))) {
+        setCheckOrigin(reportedOrigin)
+        return
+      }
+      if (reportedMismatches.has(reportedOrigin)) return
+      reportedMismatches.add(reportedOrigin)
+      console.warn('Visual Editing is here but misconfigured', {reportedOrigin})
```

Comment:
interesting, we don't really make use of console warns in the studio like this though we might want to move forward. Is this something we really want to leave available?